personalities/ct-static-api/cert_checker.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2016 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctfe
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/certificate-transparency-go/asn1"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+)
+
+var (
+	ErrNoRFCCompliantPathFound = errors.New("no RFC compliant path to root found when trying to validate chain")
+)
+
+// IsPrecertificate tests if a certificate is a pre-certificate as defined in CT.
+// An error is returned if the CT extension is present but is not ASN.1 NULL as defined
+// by the spec.
+func IsPrecertificate(cert *x509.Certificate) (bool, error) {
```

Comment:
Does this need to be exported?

I think there may be quite a lot of other funcs in here which can be unexported too.

Doesn't have to be in this PR, but in follow ups I think it would be cool to also look at how much could/should be moved e.g. into an `internal` directory.

Reply:
I've un-exported these. Alternatively, they could (should?) move to a different package (x509util?) and be re-used between c-t-go and this personality.